lib/Myriad/RPC/Client/Implementation/Memory.pm

```diff
@@ -88,14 +88,21 @@ async method call_rpc ($service, $method, %args) {
             $self->loop->timeout_future(at => $deadline),
             $pending
         );
+
+        unless (exists $message->response->{message}) {
+            my $reason = $message->response->{error}{message} // "Unknown";
+            Myriad::Exception::RPC::RemoteException->throw(reason => "Remote exception is thrown: $reason")
+                unless exists $message->response->{response};
+        }
+
         return $message->response->{response};
     } catch ($e) {
         if ($e =~ /Timeout/) {
             $e  = Myriad::Exception::RPC::Timeout->new(reason => 'deadline is due');
         } else {
-            $e = Myriad::Exception::InternalError->new(reason => $e) unless blessed $e && $e->does('Myriad::Exception');
+            $e = Myriad::Exception::InternalError->new(reason => $e) unless (ref $e) =~ /^Myriad::Exception::/;
```

Comment:
There's no guarantee the class starts with `Myriad::Exception::`, so this check is wrong - we specifically want to check that the exception *implements the Myriad::Exception rôle*.